auto_submit/lib/configuration/repository_configuration_manager.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2023 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// import 'dart:typed_data';
+
+import 'package:auto_submit/configuration/repository_configuration.dart';
+import 'package:auto_submit/service/config.dart';
+import 'package:auto_submit/service/github_service.dart';
+import 'package:auto_submit/service/log.dart';
+import 'package:github/github.dart';
+import 'package:mutex/mutex.dart';
+import 'package:neat_cache/neat_cache.dart';
+
+/// The [RepositoryConfigurationManager] is responsible for fetching and merging
+/// the autosubmit configuration from the Org level repository and if needed
+/// fetching the override configuration from the pull request repository.
+///
+/// It will attempt to access the cache first before repulling the configuraiton
+/// from the repositories. This is currently set at a 10 minute TTL.
+class RepositoryConfigurationManager {
+  final Mutex _mutex = Mutex();
+
+  static const String fileSeparator = '/';
```

Comment:
Is there a constant we can use rather than defining our own?

Reply:
Looks like there is a Platform.pathSeparator getter/constant.